### PR TITLE
Setup scorecard workflow and pin actions versions

### PR DESCRIPTION
**Pull request by mmorel-35**
Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

**Comment by mmorel-35** [comment]
cc @phlax

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8  # v3.1.0
+      with:
+        name: SARIF file
+        path: results.sarif
+        retention-days: 5
+
+    - name: "Upload to code-scanning"
+      uses: github/codeql-action/upload-sarif@012739e5082ff0c22ca6d6ab32e07c36df03c4a4  # v3.22.12
+      with:
+        sarif_file: results.sarif
```
Comment (phlax) [inline .github/workflows/scorecard.yml:45]: nit: add a newline

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -10,6 +10,6 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
       - name: build and test
         run: make docker_tests
```
Comment (phlax) [inline .github/workflows/ci.yaml:15]: same here

**Review by phlax — APPROVED** [review]
lgtm, thanks @mmorel-35

**Pull request merged**